### PR TITLE
add "~" support

### PR DESCRIPTION
**Pull request by DavidBadura**
```php
echo Path::canonicalize('~/config.ini');
// => /home/myself/config.ini
```

see #10

### CODE REVIEW

File: tests/PathTest.php [hunk 1]
```diff
@@ -1229,4 +1245,35 @@ public function testJoinFailsIfInvalidPath()
...
+
+    /**
+     * @expectedException \RuntimeException
+     * @expectedExceptionMessage Your environment or operation system isn't supported
+     */
+    public function testGetHomeDirectoryFailsIfNotSupportedOperationSystem()
+    {
+        putenv('HOME=');
+        putenv('HOMEDRIVE=');
+        putenv('HOMEPATH=');
```
Comment (webmozart) [inline tests/PathTest.php]: You should put this code into `setUp()`

File: src/Path.php [hunk 2]
```diff
@@ -89,6 +89,11 @@ public static function canonicalize($path)
             return self::$buffer[$path];
         }
 
+        // Replace "~" with user's home directory.
+        if ($path[0] === '~') {
```
Comment (webmozart) [inline src/Path.php]: Could you use YODA expressions please? e.g. `'~' === ...`

File: src/Path.php [hunk 3]
```diff
@@ -192,6 +197,39 @@ public static function getDirectory($path)
...
+        if (getenv('HOME')) {
+            return static::canonicalize(getenv('HOME'));
+        }
+
+        // For >= Windows8 support
+        if (getenv('HOMEDRIVE') && getenv('HOMEPATH')) {
+            return static::canonicalize(getenv('HOMEDRIVE') . getenv('HOMEPATH'));
+        }
+
+        throw new \RuntimeException("Your environment or operation system isn't supported");
```
Comment (webmozart) [inline src/Path.php]: Could you please import the class and use the imported class name here?

```php
use RuntimeException;

throw new RuntimeException(...);
```

**Comment by webmozart** [comment]
Looks good, thanks! :)

File: src/Path.php (hunk 3 above)
Comment (DavidBadura) [inline src/Path.php]: should I do this for the other exceptions too?

File: tests/PathTest.php (hunk 1 above)
Comment (DavidBadura) [inline tests/PathTest.php]: I have the tests improved

Comment (webmozart) [inline tests/PathTest.php]: You can also remove the duplicate statements from the individual tests now.

File: src/Path.php (hunk 3 above)
Comment (webmozart) [inline src/Path.php]: Yes, for all usages of global classes please :)

**Comment by DavidBadura** [comment]
done :)

**Comment by webmozart** [comment]
awesome, some minor nitpicks then we're good :)

File: src/Path.php [hunk 4]
```diff
@@ -89,6 +91,11 @@ public static function canonicalize($path)
             return self::$buffer[$path];
         }
 
+        // Replace "~" with user's home directory.
+        if ('~' === $path[0]) {
+            $path = static::getHomeDirectory() . substr($path, 1);
```
Comment (webmozart) [inline src/Path.php]: please remove the spaces around the `.` here and everywhere else

I'll create a PHPCS-fixer soon for stuff like that :)

File: src/Path.php [hunk 5]
```diff
@@ -504,7 +544,7 @@ public static function isRelative($path)
      *
      * @return string An absolute path in canonical form.
      *
-     * @throws \InvalidArgumentException If the base path is not absolute or if
+     * @throws InvalidArgumentException If the base path is not absolute or if
      *                                   the given path is an absolute path with
      *                                   a different root than the base path.
```
Comment (webmozart) [inline src/Path.php]: the doc block is not aligned anymore

File: tests/PathTest.php [hunk 6]
```diff
@@ -20,6 +20,26 @@
  */
 class PathTest extends \PHPUnit_Framework_TestCase
 {
+    protected $storedEnv = array();
+
+    public function setUp()
+    {
+        $this->storedEnv['HOME']      = getenv('HOME');
```
Comment (webmozart) [inline tests/PathTest.php]: please don't align the `=`. If you do, we need to change all three lines if the length of the longest line changes, which messes up the Git history

**Comment by DavidBadura** [comment]
done :)

**Comment by webmozart** [comment]
Thanks! :)

**Comment by DavidBadura** [comment]
@webmozart when you plan the next release? :)

**Comment by webmozart** [comment]
Here you go :) https://github.com/webmozart/path-util/releases/tag/2.1.0

**Comment by DavidBadura** [comment]
cool :-)

**Pull request merged**